### PR TITLE
Support PrefixedULID case class in ULID

### LINKED ISSUE #2504: ULID with prefixes
ULID with prefix (e.g., P:01AN4Z07BY79KA1307SR9X4MV3, (resource type):(ULID)) format is convenient to distinguish resource types while maintaining timestamp orders. 

For example: 
```scala
case class ResourceID(prefix:String, ulid: ULID) 
```

### PR DESCRIPTION
**Pull request by yuokada**
Close #2504

### CODE REVIEW

File: airframe-ulid/src/main/scala/wvlet/airframe/ulid/ULID.scala [hunk 1]
```diff
@@ -329,3 +329,7 @@ object ULID {
   }
 
 }
+
+case class ResourceID(prefix: String, ulid: ULID, delimiter: Char = ':') {
+  override def toString: String = s"$prefix$delimiter$ulid"
```
Comment (xerial) [inline airframe-ulid/src/main/scala/wvlet/airframe/ulid/ULID.scala:334]: Wrap individual variables with `${...}` for consistency with other code. Actually, always wrapping it is preferred for readability

File: airframe-ulid/src/main/scala/wvlet/airframe/ulid/ULID.scala [hunk 2]
```diff
@@ -329,3 +329,7 @@ object ULID {
   }
 
 }
+
+case class ResourceID(prefix: String, ulid: ULID, delimiter: Char = ':') {
```
Comment (xerial) [inline airframe-ulid/src/main/scala/wvlet/airframe/ulid/ULID.scala:333]: ResourceID is a too specific name. As a library, having a generic name is preferred. 

- PrefixedULID 
- PULID 
- etc. 

What do you think?

Comment (yuokada) [reply airframe-ulid/src/main/scala/wvlet/airframe/ulid/ULID.scala:333]: I prefer PrefixedULID. In my opinion,  it is difficult to quickly understand what `PULID` is.

**Comment by yuokada** [comment]
@xerial Fixed the code. PTAL.

**Comment by xerial** [comment]
Thanks. Create a separate class for PrefixedULID.

A general use case is using PrefixedULID as a case class parameter (in airframe-http), so in addition, we need to:
- Add a codec for PrefixedULID to serialize it as a string (See ULIDCodec as an example)
- Register the codec for PrefixedULID in the following code:
image

**Comment by xerial** [comment]
Discussion note
- Only support `:` as the delimiter
- Take the last `:` when unpack

**Comment by xerial** [comment]
@yuokada Added more test code and utility methods. PTAL

**Comment by yuokada** [comment]
@xerial Checked. Thanks for improving my code.

**Pull request merged**